.circleci/config.yml

```diff
@@ -297,115 +297,32 @@ commands:
-          name: "Restore Agent Base Docker Image Cache"
-          key: smoke-distribution-image-{{ .Branch }}-<< parameters.distribution >>-{{ checksum "~/checksum" }}-<< pipeline.parameters.cache_version >>
-
-      - run:
-          name: Copy Cached Docker Images (if they exist)
-          command: ./scripts/circleci/copy-cached-docker-images.sh
-
-      - run:
-          name: Run Unit Tests under Python << parameters.python_version >>
+          name: "Run tox target: << parameters.target_name >>"
           command: |
             # Concatenate 'CIRCLE_BUILD_NUM' and 'AGENT_HOST_NAME' env. variables.
             # when job runs from CircleCi servers, the 'AGENT_HOST_NAME' is not set, so 'CIRCLE_BUILD_NUM' is used.
             # when job runs locally, the 'CIRCLE_BUILD_NUM' is not set, so 'AGENT_HOST_NAME' is used.
             export CIRCLE_BUILD_NUM=${CIRCLE_BUILD_NUM}${AGENT_HOST_NAME}
-            export AGENT_HOST_NAME=<< parameters.agent_host_name_prefix >>-<< parameters.distribution >>-<< parameters.python_version >>-${CIRCLE_BUILD_NUM}
-
+            echo << parameters.agent_host_name >>
+            export AGENT_HOST_NAME=<< parameters.agent_host_name >>-${CIRCLE_BUILD_NUM}
+            echo "$AGENT_HOST_NAME"
             mkdir -p ~/artifacts
 
-            << parameters.command >> -- --image-cache-path ~/agent_image --artifacts-path ~/artifacts
-
-      - run:
-          name: Copy Created Docker Images to Cache Folders
-          command: ./scripts/circleci/copy-docker-images-to-cache.sh
-
-      - save_cache:
-          # save fpm builder image to 'tar' file here, if this file is not found previously.
-          name: "Save fpm Builder Base Docker Image Cache"
-          key: fpm-builder-{{ .Branch }}-{{ checksum "~/fpm_builder_checksum" }}-<< pipeline.parameters.cache_version >>
-          paths:
-            - "~/fpm_builder_cache"
-
-      - save_cache:
-          # save monitors builder image to 'tar' file here, if this file is not found previously.
-          name: "Save monitors Builder Base Docker Image Cache"
-          key: monitors-builder-{{ .Branch }}-{{ checksum "~/monitors_builder_checksum" }}-<< pipeline.parameters.cache_version >>
-          paths:
-            - "~/monitors_builder_cache"
-
-      - save_cache:
-          # save agent image to 'tar' file here, if this file is not found previously.
-          name: "Save Agent Base Docker Image Cache"
-          key: smoke-distribution-image-{{ .Branch }}-<< parameters.distribution >>-{{ checksum "~/checksum" }}-<< pipeline.parameters.cache_version >>
-          paths:
-           - "~/agent_image_cache"
-
-      - save_cache:
-          name: "Save Python Dependencies Dache"
-          key: deps2-tox-{{ .Branch }}-3.6-venv-{{ checksum "dev-requirements.txt" }}
-          paths:
-            - "~/.cache/pip"
-
+            tox -e << parameters.target_name >> -- --no-rebuild --artifacts-path ~/artifacts -n 2
```

Comment:
Can you please provide some data on how running tests in parallel affects the job timing?